probe-rs/src/config/registry.rs

```diff
@@ -544,7 +544,17 @@ mod tests {
                         raw_flash_algo
                             .cores
                             .iter()
-                            .map(|c| c.as_str())
+                            .filter_map(|c| {
+                                if target
+                                    .cores
+                                    .iter()
+                                    .any(|target_core| &target_core.name == c)
+                                {
+                                    Some(c.as_str())
+                                } else {
+                                    None
```

Comment:
Feels like this should not be silently filtered but should rather be warned about :)

Reply:
Aaaaand it's gone.

STM has some algos that are defined for both single and dual-core MCUs and the naming schemes are different. I.e. one chip ignores one core, the other ignores another.

Reply:
Hmm but then it should also not be selected, so this would not have been an issue before either, no?